### ISSUE TITLE
Username is cached for use in confirmPassword on Android

### ISSUE THREAD

**Issue by ssantomauro** [aws-amplify/amplify-flutter#464]
**Describe the bug**
Amplify.Auth.confirmPassword is not displaying any error when the email/username and the Verification Code don't match. In JS the check exists.

**To Reproduce**
1. Reset password
2. Copy the Verification Code from the email
3. Paste the Verification Code
4. Insert a wrong e-mail address

**Expected behavior**
Expected an error message

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[x] Android
[] iOS (I don't know)

**pubspec.yaml**
...
  amplify_flutter: "<1.0.0"
  amplify_auth_cognito: "<1.0.0"
...

**Comment by haverchuck** [comment]
@ssantomauro How are you trying to catch the error?

**Comment by ssantomauro** [comment]
This is my code:
```
try {
      await Amplify.Auth.confirmPassword(
          username: email,
          newPassword: newPassword,
          confirmationCode: verificationCode);
} catch (error) {
  print(error);
  throw (error);
}
```

It's the way I'm using for the signIn and signOut as well.

**Comment by haverchuck** [comment]
@ssantomauro Ok I think I found the issue.  In Android, our underlying amplify-android libray just caches the username when the initial `resetPassword` method is called (while iOS requires it to be passed).  We will work to resolve this discrepency.

**Comment by ssantomauro** [comment]
Thanks!

**Comment by Jordan-Nelson** [comment]
Hello @ssantomauro - We have recently released a new developer preview version of the Auth category ([amplify_flutter v1.0.0-next.0](https://pub.dev/packages/amplify_flutter/versions/1.0.0-next.0)). This new version is a full re-write of the auth category in dart, and it supports Mobile, Desktop, and Web. This issue is addressed in this new version. You can read more about the developer preview release in the [blog post](https://aws.amazon.com/blogs/mobile/amplify-flutter-auth-web-and-desktop/) and [docs](https://docs.amplify.aws/lib/auth/getting-started/q/platform/flutter/). I will update this issue again once this new version is out of developer preview.

Let me know if you have any questions!

**Comment by Jordan-Nelson** [comment]
This issue has been addressed in v1.0.0 of Amplify Flutter, which is now stable. This release also includes web and desktop support for Auth, API, Analytics, and Storage. You can see the list of new features and bug fixes in the [release notes](https://github.com/aws-amplify/amplify-flutter/releases/tag/v1.0.0), and see more details on how to migrate in the [upgrade guide](https://docs.amplify.aws/lib/project-setup/upgrade-guide/q/platform/flutter/).
